internal/monitors/statsd/monitor.go

```diff
@@ -0,0 +1,182 @@
+package statsd
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/sfxclient"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "statsd"
+
+var metricTypeMap = map[string]datapoint.MetricType{
+	"g":  datapoint.Gauge,
+	"c":  datapoint.Count,
+	"ms": datapoint.Gauge,
+	"s":  datapoint.Gauge,
+}
+
+var logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 30*time.Second)
+var golibLogger = &utils.LogrusGolibShim{FieldLogger: logger.FieldLogger}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// ConverterInput is to receive configs to setup metric converters
+type ConverterInput struct {
+	// A pattern to match against StatsD metric names
+	Pattern string `yaml:"pattern"`
+	// A format to compose a metric name to report to SignalFx
+	MetricName string `yaml:"metricName"`
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"false" singleInstance:"false"`
+	// The host/address on which to bind the UDP listener that accepts statsd
+	// datagrams
+	ListenAddress string `yaml:"listenAddress" default:"localhost"`
+	// The port on which to listen for statsd messages
+	ListenPort uint16 `yaml:"listenPort" default:"8125"`
+	// The protocol for statsd communications
+	ListenTCP bool `yaml:"listenTCP" default:"false"`
```

Comment:
Before we release this, you either need to implement TCP or hide this config option so that you don't tease users with it.

Reply:
I'll hide it for now. I think that was the whole point of having it as `listenTCP` and default to false.